client/rpc_sender.go

```diff
@@ -62,10 +59,15 @@ func newSender(server string, context *base.Context, retryOpts retry.Options) (*
 
 	if context.Insecure {
 		log.Warning("running in insecure mode, this is strongly discouraged. See --insecure and --certs.")
+	} else {
+		if _, err := context.GetClientTLSConfig(); err != nil {
+			return nil, err
+		}
 	}
-	ctx := roachrpc.NewContext(context, hlc.NewClock(hlc.UnixNano), nil)
-	client := roachrpc.NewClient(addr, ctx)
-	return &Sender{
+
+	ctx := rpc.NewContext(context, hlc.NewClock(hlc.UnixNano), stop.NewStopper())
```

Comment:
Is this necessary because tests previously used the http sender and are now using the rpc sender? If yes, passing the stopper to `NewSenderFunc` sounds correct.